core-bundle/src/DataContainer/DataContainerOperation.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\DataContainer;
+
+use Contao\DataContainer;
+use Contao\StringUtil;
+
+class DataContainerOperation implements \ArrayAccess
+{
+    private string|null $html = null;
+
+    /**
+     * @internal
+     */
+    public function __construct(private readonly string $name, private array $operation, private readonly array $record, private readonly DataContainer $dataContainer)
```

Comment:
Do we need `$record` now or can you use `$dataContainer->getCurrentRecord($id)`?

Reply:
I didn't want to make the object depend on a `DataContainer` object. But if we agree operations are always tied to a `$dc` then I could just pass that 🙃